src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestParser.cs

```diff
@@ -0,0 +1,412 @@
+                {                                   
+                    node.Add(ParseComment());
+                } 
+                else if (CurrentToken is { Kind: HttpTokenKind.Punctuation } and { Text: "/" } && 
+                    NextToken is { Kind: HttpTokenKind.Punctuation } and { Text: "/" })
+                {
+                    node.Add(ParseComment());
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return node;
+        }
+
+        private T ParseTrailingTrivia<T>(T node, bool stopAfterNewLine = false) where T : HttpSyntaxNode
+        {
+            while (MoreTokens())
+            {
+                if (stopAfterNewLine && CurrentToken.Kind is HttpTokenKind.NewLine)
+                {
+                    ConsumeCurrentTokenInto(node);
+                    break;
+                }
+
+                if (CurrentToken.Kind is not (HttpTokenKind.Whitespace or HttpTokenKind.NewLine))
+                {
+                    break;
+                }
+
+                ConsumeCurrentTokenInto(node);
+            }
+
+            return node;
+        }
+
+        private HttpRequestNode ParseRequest()
+        {
+            var methodNode = ParseMethod();
+            var urlNode = ParseUrl();
+            var versionNode = ParseVersion();
+            var headersNode = ParseHeaders();
+            var bodySeparatorNode = ParseBodySeparator();
+            var bodyNode = ParseBody();
+
+            return new HttpRequestNode(
+                _sourceText,
+                _syntaxTree,
+                methodNode,
+                urlNode,
+                versionNode,
+                headersNode,
+                bodySeparatorNode,
+                bodyNode);
+        }
+
+        private HttpMethodNode? ParseMethod()
+        {
+            if (MoreTokens() && CurrentToken.Kind is HttpTokenKind.Word && (CurrentToken.Text.ToLower() is "https" || CurrentToken.Text.ToLower() is "http"))
+            {
+                return null;
+            }
+
+            var node = new HttpMethodNode(_sourceText, _syntaxTree);
+
+            ParseLeadingTrivia(node);
+
+            if (MoreTokens() && CurrentToken.Kind is HttpTokenKind.Word)
+            {
+                if (CurrentToken.Text.ToLower() is ("get" or "post" or "patch" or "put" or "delete" or "head" or "options" or "trace"))
+                {                 
+                    ConsumeCurrentTokenInto(node);
+                }
+                else
+                {        
+                    var tokenSpan = _sourceText.GetSubText(CurrentToken.Span).Lines.GetLinePositionSpan(CurrentToken.Span);                
+
+                    var diagnostic = new Diagnostic(LinePositionSpan.FromCodeAnalysisLinePositionSpan(tokenSpan), DiagnosticSeverity.Warning, CurrentToken.Text.ToLower(), $"Unrecognized HTTP verb {CurrentToken.Text}");
```

Comment:
minor: Is the ToLower() call needed?

Reply:
I think it is because then GET wouldn't be recognized correct?

Reply:
ToLower makes sense in the conditions in the if block above. I am not sure it is needed when we construct the diagnostic though, is it?